Voltaire/Controllers/Messages/SendToGuild.cs

```diff
@@ -56,16 +56,27 @@ public static async Task LookupAndSendAsync(SocketGuild guild, ShardedCommandCon
                 return;
             }
 
-            if(!IncrementAndCheckMessageLimit.Perform(dbGuild, db))
+            if (!IncrementAndCheckMessageLimit.Perform(dbGuild, db))
             {
                 await Send.SendErrorWithDeleteReaction(context, "This server has reached its limit of 50 messages for the month. To lift this limit, ask an admin or moderator to upgrade your server to Voltaire Pro. (This can be done via the `!volt pro` command.)");
                 return;
             }
 
             var prefix = PrefixHelper.ComputePrefix(context, dbGuild);
             var channel = candidateChannels.OrderBy(x => x.Name.Length).First();
-            var messageFunction = Send.SendMessageToChannel(channel, replyable, context, dbGuild.UseEmbed);
-            await messageFunction(prefix, message);
+            if (context.Message.Attachments.Any())
```

Comment:
You're right that this would be preferable, and it was in fact how I did it initially. It worked fine if a message had a single attachment, but when there were multiple, I noticed that the whole `Func<string, string, Task<IUserMessage>>` function is supposed to return only a single `IUserMessage` Task, and I couldn't really figure out a way around that (as the messageFunction is `awaited` all the way back in `SendToGuild`). 
Doing the logic outside of the Async area was the easiest way I could find to make sense of it.

Reply:
Right - could we, for example, return a function that returns that last IUserMessage? Or an array of them?

Reply:
as they are awaited inside of `SendToGuild`, we can't really just return the last one. It would have to be a list. 
I'm not very familiar with Async flows so I'm not sure if it can be an ordinary list or if it has to be some kind of Async List.
My issue with using an Async List is that the attachments might have their order jumbled, which would be a problem if the message text was only attached to the first file.
If we do go that route, `Task.WhenAll()` [should](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenall?view=netcore-3.1#System_Threading_Tasks_Task_WhenAll_System_Collections_Generic_IEnumerable_System_Threading_Tasks_Task__) be enough.

Reply:
Yeah - the caller is what is doing the awaiting, but I think all of the callers are essentially just awaiting the returned message. We may have to alter the inner workings, but we should be able to have this thing await the returned function and have it receive the result of sending the last message, I think.

I'm happy to take a whack at this at this week some point 🙂.

Reply:
I'd appreciate that, I'm struggling a little to get my head around async, hence the current state of my changes.

Reply:
Oh, one more issue with just returning the last message;
sendReply and SendDirectMessage both add the Delete React to each message that gets sent. if you look at those, the `Send.AddReactionToMessage` is inside the loop, to let users delete each image individually.